deltachat tests::verified_chats::test_verified_chat_editor_reordering is flaky

Recent failure from macOS CI:
```
    test tests::verified_chats::test_verified_chat_editor_reordering ... FAILED

    failures:

    failures:
        tests::verified_chats::test_verified_chat_editor_reordering

    test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 994 filtered out; finished in 0.68s

  stderr ───

    thread 'tests::verified_chats::test_verified_chat_editor_reordering' panicked at src/test_utils.rs:1009:13:
    assertion failed: `(left == right)`: To update the expected value, run `UPDATE_GOLDEN_TESTS=1 cargo test`

    Diff < left / right > :
     Group#Chat#3002: Group [3 member(s)]
     --------------------------------------------------------------------------------
     Msg#3004: info (Contact#Contact#Info): alice@example.org invited you to join this group.
     
     Waiting for the device of alice@example.org to reply… [NOTICED][INFO]
    <Msg#3002: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
     Msg#3006: info (Contact#Contact#Info): alice@example.org replied, waiting for being added to the group… [NOTICED][INFO]
    >Msg#3002: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
     Msg#3008🔒:  (Contact#Contact#3002):  [FRESH]
     Msg#3009: info (Contact#Contact#Info): Member bob@example.net added. [NOTICED][INFO]
     Msg#3010🔒:  (Contact#Contact#3001): Member Me added by alice@example.org. [FRESH][INFO]
     --------------------------------------------------------------------------------
     


    stack backtrace:
       0: __rustc::rust_begin_unwind
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/std/src/panicking.rs:697:5
       1: core::panicking::panic_fmt
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/panicking.rs:75:14
       2: deltachat::test_utils::TestContext::golden_test_chat::{{closure}}
                 at ./src/test_utils.rs:1009:13
       3: deltachat::tests::verified_chats::test_verified_chat_editor_reordering::{{closure}}
                 at ./src/tests/verified_chats.rs:713:10
       4: <core::pin::Pin<P> as core::future::future::Future>::poll
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/future/future.rs:133:9
       5: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/park.rs:284:71
       6: tokio::task::coop::with_budget
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/task/coop/mod.rs:167:5
       7: tokio::task::coop::budget
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/task/coop/mod.rs:133:5
       8: tokio::runtime::park::CachedParkThread::block_on
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/park.rs:284:31
       9: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/context/blocking.rs:66:14
      10: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/scheduler/multi_thread/mod.rs:87:22
      11: tokio::runtime::context::runtime::enter_runtime
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/context/runtime.rs:65:16
      12: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/scheduler/multi_thread/mod.rs:86:9
      13: tokio::runtime::runtime::Runtime::block_on_inner
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/runtime.rs:358:50
      14: tokio::runtime::runtime::Runtime::block_on
                 at /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/runtime.rs:330:18
      15: deltachat::tests::verified_chats::test_verified_chat_editor_reordering
                 at ./src/tests/verified_chats.rs:714:7
      16: deltachat::tests::verified_chats::test_verified_chat_editor_reordering::{{closure}}
                 at ./src/tests/verified_chats.rs:646:52
      17: core::ops::function::FnOnce::call_once
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/ops/function.rs:253:5
      18: core::ops::function::FnOnce::call_once
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/ops/function.rs:253:5
    note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
    note: If you want to examine the database files, set environment variable DELTACHAT_SAVE_TMP_DB=1

  Cancelling due to test failure: 2 tests still running
        PASS [   0.157s] ( 903/1032) deltachat tests::verified_chats::test_verified_oneonone_chat_not_broken_by_classical
        PASS [   0.186s] ( 904/1032) deltachat tests::verified_chats::test_verified_oneonone_chat_not_broken_by_device_change
────────────
     Summary [  54.931s] 904/1032 tests run: 903 passed, 1 failed, 1 skipped
        FAIL [   0.695s] deltachat tests::verified_chats::test_verified_chat_editor_reordering
warning: 128/1032 tests were not run due to test failure (run with --no-fail-fast to run all tests, or run with --max-fail)
error: test run failed
Error: Process completed with exit code 100.
```

https://github.com/chatmail/core/actions/runs/18777462449/job/53575288879?pr=7348

Same for `receive_imf::receive_imf_tests::test_two_group_securejoins`, it also got flaky in CI recently:
```
    test receive_imf::receive_imf_tests::test_two_group_securejoins ... FAILED

    failures:

    failures:
        receive_imf::receive_imf_tests::test_two_group_securejoins

    test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 994 filtered out; finished in 0.86s

  stderr ───

    thread 'receive_imf::receive_imf_tests::test_two_group_securejoins' panicked at src/test_utils.rs:1009:13:
    assertion failed: `(left == right)`: To update the expected value, run `UPDATE_GOLDEN_TESTS=1 cargo test`

    Diff < left / right > :
     Group#Chat#6002: Group [3 member(s)]
     --------------------------------------------------------------------------------
     Msg#6004: info (Contact#Contact#Info): alice@example.org invited you to join this group.
     
     Waiting for the device of alice@example.org to reply… [NOTICED][INFO]
    <Msg#6002: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
     Msg#6006: info (Contact#Contact#Info): alice@example.org replied, waiting for being added to the group… [NOTICED][INFO]
    >Msg#6002: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
     Msg#6008🔒:  (Contact#Contact#6001): Member Me added by alice@example.org. [FRESH][INFO]
     --------------------------------------------------------------------------------
     


    stack backtrace:
       0: __rustc::rust_begin_unwind
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/std/src/panicking.rs:697:5
       1: core::panicking::panic_fmt
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/panicking.rs:75:14
       2: deltachat::test_utils::TestContext::golden_test_chat::{{closure}}
                 at ./src/test_utils.rs:1009:13
       3: deltachat::receive_imf::receive_imf_tests::test_two_group_securejoins::{{closure}}
                 at ./src/receive_imf/receive_imf_tests.rs:5099:10
       4: <core::pin::Pin<P> as core::future::future::Future>::poll
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/future/future.rs:133:9
       5: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/park.rs:284:71
       6: tokio::task::coop::with_budget
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/task/coop/mod.rs:167:5
       7: tokio::task::coop::budget
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/task/coop/mod.rs:133:5
       8: tokio::runtime::park::CachedParkThread::block_on
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/park.rs:284:31
       9: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/context/blocking.rs:66:14
      10: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/scheduler/multi_thread/mod.rs:87:22
      11: tokio::runtime::context::runtime::enter_runtime
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/context/runtime.rs:65:16
      12: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/scheduler/multi_thread/mod.rs:86:9
      13: tokio::runtime::runtime::Runtime::block_on_inner
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/runtime.rs:358:50
      14: tokio::runtime::runtime::Runtime::block_on
                 at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.45.1/src/runtime/runtime.rs:330:18
      15: deltachat::receive_imf::receive_imf_tests::test_two_group_securejoins
                 at ./src/receive_imf/receive_imf_tests.rs:5101:7
      16: deltachat::receive_imf::receive_imf_tests::test_two_group_securejoins::{{closure}}
                 at ./src/receive_imf/receive_imf_tests.rs:5080:42
      17: core::ops::function::FnOnce::call_once
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/ops/function.rs:253:5
      18: core::ops::function::FnOnce::call_once
                 at /rustc/1159e78c4747b02ef996e55082b704c09b970588/library/core/src/ops/function.rs:253:5
    note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
    note: If you want to examine the database files, set environment variable DELTACHAT_SAVE_TMP_DB=1

  Cancelling due to test failure: 3 tests still running
        PASS [   0.234s] ( 781/1032) deltachat receive_imf::receive_imf_tests::test_wrong_date_in_imf_section
        PASS [   0.081s] ( 782/1032) deltachat receive_imf::receive_imf_tests::test_xing_mailing_list
        PASS [   0.529s] ( 783/1032) deltachat receive_imf::receive_imf_tests::test_weird_and_duplicated_filenames
────────────
     Summary [  43.224s] 783/1032 tests run: 782 passed, 1 failed, 1 skipped
        FAIL [   0.891s] deltachat receive_imf::receive_imf_tests::test_two_group_securejoins
warning: 249/1032 tests were not run due to test failure (run with --no-fail-fast to run all tests, or run with --max-fail)
error: test run failed
Error: Process completed with exit code 100.
```

--- a/src/securejoin/bob.rs
+++ b/src/securejoin/bob.rs
@@ -65,11 +65,26 @@ pub(super) async fn start_protocol(context: &Context, invite: QrInvite) -> Resul
             )
             .await?;
 
-        // Chat ID of the group we are joining, unused otherwise.
-        let group_chat_id = joining_chat_id(context, &invite, chat_id).await?;
-        if matches!(invite, QrInvite::Group { .. })
-            && is_contact_in_chat(context, group_chat_id, ContactId::SELF).await?
-        {
+        // `group_chat_id` is `Some` if group chat
+        // already exists and we are in the chat.
+        let group_chat_id = match invite {
+            QrInvite::Group { ref grpid, .. } => {
+                if let Some((group_chat_id, _blocked)) =
+                    chat::get_chat_id_by_grpid(context, grpid).await?
+                {
+                    if is_contact_in_chat(context, group_chat_id, ContactId::SELF).await? {
+                        Some(group_chat_id)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            }
+            QrInvite::Contact { .. } => None,
+        };
+
+        if let Some(group_chat_id) = group_chat_id {
             // If QR code is a group invite
             // and we are already in the chat,
             // nothing needs to be done.

--- a/test-data/golden/two_group_securejoins
+++ b/test-data/golden/two_group_securejoins
@@ -4,6 +4,6 @@ Msg#6004: info (Contact#Contact#Info): alice@example.org invited you to join thi
 
 Waiting for the device of alice@example.org to reply… [NOTICED][INFO]
 Msg#6006: info (Contact#Contact#Info): alice@example.org replied, waiting for being added to the group… [NOTICED][INFO]
-Msg#6002: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
+Msg#6003: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
 Msg#6008🔒:  (Contact#Contact#6001): Member Me added by alice@example.org. [FRESH][INFO]
 --------------------------------------------------------------------------------

--- a/test-data/golden/verified_chats_editor_reordering
+++ b/test-data/golden/verified_chats_editor_reordering
@@ -4,7 +4,7 @@ Msg#3004: info (Contact#Contact#Info): alice@example.org invited you to join thi
 
 Waiting for the device of alice@example.org to reply… [NOTICED][INFO]
 Msg#3006: info (Contact#Contact#Info): alice@example.org replied, waiting for being added to the group… [NOTICED][INFO]
-Msg#3002: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
+Msg#3003: info (Contact#Contact#Info): Messages are end-to-end encrypted. [NOTICED][INFO]
 Msg#3008🔒:  (Contact#Contact#3002):  [FRESH]
 Msg#3009: info (Contact#Contact#Info): Member bob@example.net added. [NOTICED][INFO]
 Msg#3010🔒:  (Contact#Contact#3001): Member Me added by alice@example.org. [FRESH][INFO]

